assets/js/modules/idea-hub/util/module.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Idea Hub Module utilities.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { CORE_MODULES } from '../../../googlesitekit/modules/datastore/constants';
+
+/**
+ * Checks whether the idea hub module is connected or not.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} registry Datastore registry.
+ * @return {boolean} Returns true if the module is connected, otherwise false.
+ */
+export const isIdeaHubModuleConnected = async ( registry ) => {
```

Comment:
I like the idea of a utility to encapsulate the odd requirement to wait for `getModules` but ideally we should be able to simply `await __experimentalResolveSelect( CORE_MODULES ).isModuleConnected( 'idea-hub' )`. The only reason we can't is because `isModuleConnected` doesn't have a resolver.

Rather than doing this here, I'll open a new issue for making this change. It may be a while until that lands though, so in the meantime, let's handle this a little differently in a non-module specific way:

Let's move this to a new `experimentalIsModuleConnected( slug )` function on `Widgets` which already has `registry` in scope. This can then do the same thing it does here, but take the `slug` as an argument. This way, we also don't actually need to add `registry` on to the `widgets` instance yet as it's only needed for this right now. This way other modules can use it too but we can still consider it experimental for now and solidify it later if we decide to keep it.

Reply:
🤦 Correct. Removed!